sdk/src/main/java/io/radar/sdk/RadarSettings.kt

```diff
@@ -311,15 +313,41 @@ internal object RadarSettings {
         return RadarFeatureSettings.fromJson(JSONObject(optionsJson))
     }
 
+    fun setSdkConfiguration(context: Context, configuration: RadarSdkConfiguration?) {
+        Radar.logger.d("set SDK Configuration | sdkConfiguration = $configuration")
+        if (configuration != null) {
+            getSharedPreferences(context).edit { putInt(KEY_LOG_LEVEL, configuration.logLevel.value) }
+        }
+    }
+
+    fun getClientSdkConfiguration(context: Context): JSONObject {
+        val sharedPrefClientSdkConfig = getSharedPreferences(context).getString(KEY_CLIENT_SDK_CONFIGURATION, null);
+
+        return if (sharedPrefClientSdkConfig != null) {
+            JSONObject(sharedPrefClientSdkConfig)
+        } else {
+            JSONObject()
+        }
+    }
+
     internal fun getLogLevel(context: Context): Radar.RadarLogLevel {
         val logLevelInt = getSharedPreferences(context).getInt(KEY_LOG_LEVEL, 3)
         val userDebug = getUserDebug(context)
         return if (userDebug) Radar.RadarLogLevel.DEBUG else Radar.RadarLogLevel.fromInt(logLevelInt)
     }
 
     internal fun setLogLevel(context: Context, level: Radar.RadarLogLevel) {
-        val logLevelInt = level.value
-        getSharedPreferences(context).edit { putInt(KEY_LOG_LEVEL, logLevelInt) }
+        val sdkConfiguration = getClientSdkConfiguration(context)
```

Comment:
echoing my sentiments from the ios PR, maybe move the syncing logic to the logger obj?

Reply:
https://github.com/radarlabs/radar-sdk-android/pull/364